ClientApp/src/components/ListingsComp.js

```diff
@@ -4,23 +4,65 @@ import { ListingComp } from './ListingComp';
 export const ListingsComp = () => {
   const [loading, setLoading] = useState(true);
   const [listingsData, setListingsData] = useState([]);
+  const [searchTerm, setSearchTerm] = useState('');
   
   const fetchData = async () => {
     const response = await fetch('listing');
     const data = await response.json();
     setListingsData(data);
     setLoading(false);
   };
-
   useEffect(() => {
     fetchData();
   }, []);
 
+    const sortArray = type => {
+        switch (type) {
+            case 'dateDesc':
+                setListingsData(
+                    [...listingsData].sort((a, b) => {
+                        return new Date(b['datePublished']) - new Date(a['datePublished'])
+                    })
+                )
+                break;
+            case 'dateAsc':
+                setListingsData(
+                    [...listingsData].sort((a, b) => {
+                        return new Date(b['datePublished']) - new Date(a['datePublished'])
+                    }).reverse()
+                )
+                break;
+            default:
+                break;
+        }
+    }
+
   return (
-    <div>
-      {loading
-        ? <p><em>Loading...</em></p>
-        : <ul> {listingsData.map(data => <li key={data.id}><ListingComp {...data} refresh={fetchData} /></li>)} </ul>
+      <div style={{ overflow: 'hidden' }}>
+      {loading ? <p><em>Loading...</em></p>
+              : <><input
+                  type="text"
+                  placeholder="Search..."
+                  onChange={(event) => {
+                      setSearchTerm(event.target.value);
+                  } }
+              />
+                  <select onChange={(e) => sortArray(e.target.value)}>
+                      <option>Sort by...</option>
```

Comment:
You could add <option selected disabled> so that it would not be possible to select sort by when something else is selected.